### ISSUE TITLE
Amazing library!

### ISSUE THREAD

**Issue by sbarthelmess** [hierynomus/smbj#391]
All I ask is that you please add a little more to your documentation - this should be the leading SMB library for Android!  It's head-and-shoulders above the latest JCIFS library I struggled with for over a week (due to no SMBv2 auth support, incomplete examples, outdated and sparse docs).

To get it to compile in the latest Android Studio (3.x), just add this to your gradle:
```
dependencies {
    implementation 'com.hierynomus:smbj:0.8.0'
    implementation 'org.slf4j:slf4j-api:1.7.25'
}
```
And if you could add a code snippet to copy files to and from an SMB share I think you'd round out the use-cases to make this the de-facto standard lib.  I can certainly help if you need some examples.  Regardless, thanks for a kick butt library and saving me a TON of time!

-=poSeidon=-

**Comment by TheRealHypo** [comment]
I support that ! 

But a little more in depth examples or explanation in Code would be perfect and is obviously missing so far. Takes a while currently to adapt our JCIFS Implementation to this library.

**Comment by fblampe** [comment]
I agree, it would be very helpful to add a documentation section showing how to open, read and write a file on a share. 
Maybe this comment can serve as a reference? https://github.com/hierynomus/smbj/issues/483#issuecomment-504851114
